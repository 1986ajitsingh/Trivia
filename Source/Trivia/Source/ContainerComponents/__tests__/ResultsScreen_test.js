import mockStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import React from 'react';
import ModuleUnderTest from '../ResultsScreen';

jest.mock('react-redux', () => ({
  connect: () => (type => (type)),
}));

const initialState = {};
const store = mockStore(initialState);

describe('renders correctly', () => {
  /* eslint-disable no-undef */
  test('normal', () => {
    /* eslint-enable no-undef */
    // Arrange/ Act
    const mockCalculateTotalCorrectAnswers = jest.fn();
    const tree =
      renderer.create(<ModuleUnderTest
        calculateTotalCorrectAnswers={mockCalculateTotalCorrectAnswers}
      />).toJSON();

    // Assert
    expect(tree).toMatchSnapshot();
    expect(mockCalculateTotalCorrectAnswers).toBeCalled();
  });
});

describe('componentWillMount', () => {
  /* eslint-disable no-undef */
  test('normal', () => {
    /* eslint-enable no-undef */
    // Arrange
    const mockCalculateTotalCorrectAnswers = jest.fn();
    const moduleUnderTest = new ModuleUnderTest(
      { },
      { store },
    );
    moduleUnderTest.props = {
      calculateTotalCorrectAnswers: mockCalculateTotalCorrectAnswers,
    };

    // Act
    moduleUnderTest.componentWillMount();

    // Assert
    expect(mockCalculateTotalCorrectAnswers).toBeCalled();
  });
});

describe('componentWillReceiveProps', () => {
  [
    [1, '1. questions.length === 0', { questions: [] }],
    [2, '2. questions.length !== 0', { questions: [1, 2] }],
  ].forEach((testcase) => {
    const [scenario, testname, nextProps] = testcase;
    /* eslint-disable no-undef */
    test(testname, () => {
    /* eslint-enable no-undef */
    // Arrange
      const mockNavigate = jest.fn();
      const moduleUnderTest = new ModuleUnderTest(
        { },
        { store },
      );
      moduleUnderTest.props = {
        navigation: {
          navigate: mockNavigate,
        },
      };

      // Act
      moduleUnderTest.componentWillReceiveProps(nextProps);

      // Assert
      switch (scenario) {
        case 1:
          expect(mockNavigate).toBeCalledWith('Home');
          break;
        case 2:
          expect(mockNavigate).not.toBeCalled();
          break;
        default:
          break;
      }
    });
  });
});

describe('onPressPlayAgain', () => {
  /* eslint-disable no-undef */
  test('normal', () => {
    /* eslint-enable no-undef */
    // Arrange
    const mockResetQuestions = jest.fn();
    const moduleUnderTest = new ModuleUnderTest(
      { },
      { store },
    );
    moduleUnderTest.props = {
      resetQuestions: mockResetQuestions,
    };

    // Act
    moduleUnderTest.onPressPlayAgain();

    // Assert
    expect(mockResetQuestions).toBeCalled();
  });
});
