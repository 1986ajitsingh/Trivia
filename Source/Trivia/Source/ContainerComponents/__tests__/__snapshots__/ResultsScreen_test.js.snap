// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly normal 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E6E6FA",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingVertical": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#4B0082",
        "fontSize": 34,
        "fontWeight": "bold",
        "marginTop": 20,
        "textAlign": "center",
      }
    }
  >
    You scored 
    

     
     / 
    0
  </Text>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#4B0082",
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      PLAT AGAIN?
    </Text>
  </View>
</View>
`;
